hw/opentitan/ot_hmac.c

```diff
@@ -612,19 +854,51 @@ static void ot_hmac_regs_write(void *opaque, hwaddr addr, uint64_t value,
             break;
         }
 
-        s->regs->cfg =
-            val32 &
+        val32 &=
             (R_CFG_HMAC_EN_MASK | R_CFG_SHA_EN_MASK | R_CFG_ENDIAN_SWAP_MASK |
              R_CFG_DIGEST_SWAP_MASK | R_CFG_KEY_SWAP_MASK |
              R_CFG_DIGEST_SIZE_MASK | R_CFG_KEY_LENGTH_MASK);
 
+        /* If the digest size is invalid, it gets mapped to SHA2_NONE. */
+        digest_size = ot_hmac_get_digest_size(val32);
+        if (digest_size == HMAC_SHA2_NONE) {
+            val32 &= ~R_CFG_DIGEST_SIZE_MASK;
+            val32 |= 0x8u << R_CFG_DIGEST_SIZE_SHIFT;
```

Comment:
Would be nice to use some `#define` constants here (for `SHA2_NONE` and `KEY_NONE`)

Reply:
Can you clarify what exactly you mean here? Are you referring to making manifest constants for the `0x8u << R_CFG_DIGEST_SIZE_SHIFT` and `0x20u << R_CFG_KEY_LENGTH_SHIFT` values, or for something else?

Reply:
Ah, apologies, I think I misunderstood - are you referring to using e.g. `FIELD_DP32(storage, reg, field, val)`?

Reply:
Something like
```c

// somewhere after the REG32 section
#define CFG_DIGEST_SHA2_NONE 0x8u
#define CFG_KEY_LENGTH_NONE  0x20u

// ...

val32 |= CFG_DIGEST_SHA2_NONE << R_CFG_DIGEST_SIZE_SHIFT;
// ...
val32 |= CFG_KEY_LENGTH_NONE << R_CFG_KEY_LENGTH_SHIFT;
```

Reply:
> Ah, apologies, I think I misunderstood - are you referring to using e.g. `FIELD_DP32(storage, reg, field, val)`?

No, `val32 |= ...` is fine.